### ISSUE TITLE
Please add "install" hook

### ISSUE THREAD

**Issue by jacekn** [snapcore/snapd#2594]
I am snapping server software that needs configuration files to run. I have 2 requirements:
1. Users must be allowed to edit config files
2. "snap install <mysnap>" needs to results in a running daemon that uses sensible default config

Currently I can achieve this using wrapper script that checks whether config files exists in $SNAP_DATA and if not it copies them in place right before my software starts.

It would be much better if I could use built in method to ship config files, I think "install" hook or similar that runs only once when snap is installed would help me.

Could "install" hook be added?

**Comment by niemeyer** [comment]
We have  a "configure" hook that runs also at install time, but indeed it runs only after the snap is already setup to run (dameons on, etc). So far we considered an "install" hook redundant for that reason, but maybe it'd make sense to have something that runs before the daemons are up.

**Comment by cyberb** [comment]
We had to add the support in our fork for exactly the same reason, but sadly Canonical thinks it is redundant: 
https://github.com/syncloud/snapd/blob/master/overlord/snapstate/snapstate.go#L140
Example:
https://github.com/syncloud/platform/blob/master/snap/meta/hooks/post-install

**Comment by niemeyer** [comment]
@cyberb My comment above actually provides some more context for this, and hopefully makes clear that this is not settled on stone.

**Comment by battlemidget** [comment]
We also use the configure hook for setting up things like iptables rules, would be nice to have maybe a **remove** hook as well?

**Comment by zyga** [comment]
This issue has been moved to launchpad: https://bugs.launchpad.net/snapd/+bug/1657106
